### DISCUSSION TITLE
更新日志 - 22.03.6 - 20240115

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1264] [Announcements]
此版本为临时更新，为了解决上一版本 x86 RTL8125 网卡死锁的问题。如果不是 RTL8125 用户且已经升级 20240112，请忽略此版本。

**Comment by maisimon2000** [comment]
passwall插件负载均衡失效了

**Reply by maisimon2000** [reply]
22.03.6  之后就失效了

**Reply by jjm2473** [reply]
重装haproxy试试

**Reply by jjm2473** [reply]
软件包那边点一下更新列表，然后搜索haproxy，卸载重装。可能之后要重装下passwall。

**Reply by bcseputetto** [reply]
passwall的haproxy(负载均衡)问题详细看最新更新日志的三楼
https://github.com/istoreos/istoreos/discussions/1281#discussioncomment-8189457

**Comment by xmasrain** [comment]
![image](https://github.com/istoreos/istoreos/assets/58833842/19beb540-0576-40fe-b5d4-3a8aa9962bee)
更新后，固件检查失败

**Reply by jjm2473** [reply]
一般都是固件服务器的下载节点不稳定，过几分钟试试

**Comment by lwj1994** [comment]
什么时候修复一下，新安装系统  192.168.100.1 无法登录后台

**Reply by jjm2473** [reply]
先搜索issue，如果没有，就开个新issue，提供详细的信息

**Reply by lwj1994** [reply]
https://github.com/istoreos/istoreos/issues/1195  这个得高优安排下

**Reply by jjm2473** [reply]
如果是单网口的机器，默认是dhcp客户端，不是固定ip的

**Comment by kasen-zhang** [comment]
想知道在首页进行固件升级是否会影响以前装的插件和docker，更新完需要重装插件、docker，重新配置么？

**Reply by jjm2473** [reply]
docker不需要，插件的话，配置不需要，但是看情况可能有些插件要重新安装或者升级，例如nginx，haproxy

**Reply by kasen-zhang** [reply]
非常感谢，我目前有的插件是：alist,rclone,openclash,push-bot,aria2,ddns-go,pgyvpn,wireguard,samba，大佬这些需要重新安装么，重新安装了是否意味着需要重新配置，重新安装倒是没啥，主要有些配置折腾起来太麻烦了

**Reply by jjm2473** [reply]
大部分不需要重新安装，就算重新安装，也不需要重新配置（除非插件设计有bug）。aria2可能要升级，到时候你看能不能启动，能启动就不用升级。pgyvpn我也不知道啥，同样也是看能不能启动，不能启动就重新安装或升级。

**Reply by kasen-zhang** [reply]
我已经更新了，目前所有已装并配置的插件及docker 均正常，其中aria2 和pgyvpn也没问题。

**Comment by 30010252** [comment]
passwall插件负载均衡失效

**Reply by jjm2473** [reply]
前面不是回复了吗？

**Comment by kasen-zhang** [comment]
不知道目前这个版本有没有快照功能，是否后期有考虑增加

**Reply by jjm2473** [reply]
又不是虚拟机，哪来的快照。你用过沙箱没？

**Reply by kasen-zhang** [reply]
没有用过沙箱，我担心哪天系统奔溃又要重新安装配置插件和dockers，配置起来挺麻烦的

**Reply by jjm2473** [reply]
docker怎么也不需要配置啊，其他的配置可以备份的

**Comment by wionch** [comment]
haproxy重装, passwall重装
还是不能运行. 
负载均衡那里显示进入web, 点击还是报错
再次重新安装haproxy和passwall以后, 进入web按钮就消失了. 怎么重装也没用了.
这个版本应该是有问题的
希望官网尽快修复吧. 谢谢

**Reply by jjm2473** [reply]
看看日志。还有，你有没有使用第三方的软件源？（首页的软件源镜像除外）你的passwall是从哪下载的

**Reply by wionch** [reply]
passwall从 `https://github.com/AUK9527/Are-u-ok/tree/main/x86` 下载的

**Reply by wionch** [reply]
没有使用第三方的软件源.
haproxy的安装日志:
```
Installing haproxy (2.4.22-1) to root...
Downloading https://mirrors.tuna.tsinghua.edu.cn/openwrt/releases/22.03.6/packages/x86_64/packages/haproxy_2.4.22-1_x86_64.ipk
Installing libltdl7 (2.4.6-2) to root...
Downloading https://mirrors.tuna.tsinghua.edu.cn/openwrt/releases/22.03.6/packages/x86_64/base/libltdl7_2.4.6-2_x86_64.ipk
Installing liblua5.3-5.3 (5.3.5-5) to root...
Downloading https://mirrors.tuna.tsinghua.edu.cn/openwrt/releases/22.03.6/packages/x86_64/base/liblua5.3-5.3_5.3.5-5_x86_64.ipk
Configuring libltdl7.
Configuring liblua5.3-5.3.
Configuring haproxy.
```
状态-系统日志里关于haproxy的日志:
```
Fri Jan 19 11:49:32 2024 daemon.info procd: Instance haproxy::instance1 s in a crash loop 6 crashes, 0 seconds since last crash
```
其他就没看到什么日志了.

**Reply by futianyi1994** [reply]
我的是负载均衡显示运行中但是就是进不了web管理，负载均衡节点一直超时用不了。不知道啥情况。卸载重装了几次还是一样

**Reply by wionch** [reply]
我的也是, 弄了一下午了, 也是显示运行中, 进不了web, 点击测试显示超时
安装老版本的passwall也是同样的.

但是我另外安装passwall2, 里面的负载均衡可以正常使用.

奇葩的问题.
希望官方能解决下这个问题.

**Reply by jjm2473** [reply]
https://github.com/AUK9527/Are-u-ok/tree/main/x86 可能已经不支持 22.03.6，可以试试  https://github.com/bcseputetto/Are-u-ok/releases

**Reply by wionch** [reply]
我这个问题已经解决了.
把passwall和haproxy重装
再把负载均衡里的记录清空
再重新设置一次负载均衡就ok了

**Reply by jjm2473** [reply]
负载均衡列表应该不需要清理，你要看看负载均衡到底是什么模式的，“健康检测类型”如果是“passwall内置xxx”，那么添加负载均衡节点的时候要选择socks协议（或者xray的socks协议，但是要把所有参数都改成原始类型，不能使用tls，websocket，混淆等），如果“健康检测类型”是“TCP”，那就按照以前的方式配置。

详情看passwall官方的讨论 https://github.com/xiaorouji/openwrt-passwall/discussions/2420#discussioncomment-5473163

**Reply by wionch** [reply]
好de, 多谢

**Reply by futianyi1994** [reply]
负载均衡控制台始终进不去重装了好多次，现在还出了其他问题节点类型v2ray没了

**Reply by bcseputetto** [reply]
@futianyi1994 passwall从4.70-1开始就不再依赖v2ray，直接使用xray或者sing-box来代替v2ray跑vless或者vmess。负载均衡的问题注意是否配置错误，尝试一下不添加任何负载均衡节点的情况下启用负载均衡能否正常启动并打开负载均衡后台，如果可以说明负载均衡没有问题，注意添加负载均衡节点时不要重复添加。

**Reply by bcseputetto** [reply]
顺便一提，负载均衡不需要挑协议参数一样的节点，新建负载均衡节点时也不需要写多余的参数，直接一个socks节点127.0.0.1，端口1181指向haproxy监听的地址端口即可。

**Reply by futianyi1994** [reply]
不添加负载均衡节点的情况下，负载节点显示未运行，后台还是打不开。添加负载均衡节点就显示运行中但是还是打不开后台。负载均衡对节点有什么要求吗我是ssr

**Reply by wionch** [reply]
按顺序重装haproxy和passwall
然后, 清空所有负载均衡的节点
重新添加负载节点
尝试用这个顺序执行一次看

 然后按照@bcseputetto说的: "新建负载均衡节点时也不需要写多余的参数，直接一个socks节点127.0.0.1，端口1181指向haproxy监听的地址端口即可"
亲测有效
我就是这么解决的

**Reply by futianyi1994** [reply]
是先依次卸载hapoxy和passwall，在依次重装

**Reply by futianyi1994** [reply]
可是我这边卸载passwall或者卸载haproxy任意一个另外一个也没了，装上passwall后haproxy也已经装上了

**Reply by bcseputetto** [reply]
@futianyi1994 麻烦你打开终端，或者用ssh登录后，执行haproxy，给我截图个
![image](https://github.com/istoreos/istoreos/assets/29139573/8ea78c1b-342e-43d3-96dd-d44b166e7135)

**Reply by futianyi1994** [reply]
![IMG_20240127_131054](https://github.com/istoreos/istoreos/assets/23741206/14ca8bab-38a8-4bc8-b842-253a6de105b3)

**Reply by bcseputetto** [reply]
@futianyi1994 请把负载均衡下面的节点清空
![image](https://github.com/istoreos/istoreos/assets/29139573/51f95bf1-2aed-4652-801e-8175b0b35827)
然后输入账号密码，启用，看下后台能不能打开

**Reply by futianyi1994** [reply]
清空了都没后台入口。直接输地址也打不开，负载均衡显示未运行

**Reply by futianyi1994** [reply]
![Uploading Screenshot_2024-01-27-13-35-41-253_com.microsoft.emmx.png…]()

**Reply by bcseputetto** [reply]
@futianyi1994 给你126邮箱发了联系方式，请联系我

**Reply by futianyi1994** [reply]
解决了，是DNS配置错误导致haproxy无法启动

**Reply by jjm2473** [reply]
passwall新的负载均衡配置方式参考官方的讨论区 https://github.com/xiaorouji/openwrt-passwall/discussions/2420#discussioncomment-5473163



PS：如果还想用以前的方式配置（也就是协议加密参数匹配），那么把负载均衡的健康检测算法改成TCP，就是以前的方式了（我也不知道为啥叫健康检测算法）
